MWT/how-to-container.rst

```diff
@@ -1,6 +1,66 @@
 How-To Create a Container
 =========================
 
+To create a custom container, a new class should be created, extending the ``Container`` class.
+This new class may define a constructor and setter methods in order to provide a way for the user to configure the container, such as its orientation.
+Container subclasses have to implement two methods and may override optional methods, as explained in the following sections.
+
+Implementing the mandatory methods
+----------------------------------
+
+This section explains how to implement the two mandatory methods of a container subclass.
+
+Computing the optimal size of the container
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The ``computeContentOptimalSize()`` method is called by the MWT framework in order to know the optimal size of the container.
+The optimal size of the container should be big enough so that each child can be allocated a size at least as big as its own optimal size.
+For example, the optimal width of a horizontal list is the sum of the optimal width of every child, and the optimal height is the maximum optimal height of all children.
+
+The container is responsible for computing the optimal size of every child. To do so, the ``computeChildOptimalSize()`` method should be called for every child.
+After this method is called, the optimal size of the child can be retrieved by calling ``getWidth()`` and ``getHeight()`` on the child widget.
+
+The ``Size`` parameter of the ``computeContentOptimalSize()`` method initially contains the size available for the container.
+An available width or height equal to ``Widget.NO_CONSTRAINT`` means that the optimal size should be computed without considering any restriction on the respective axis.
+Before the method returns, the size object should be set to the optimal size of the container.
+
+"blurry point": should the optimal size always be lower than the available size?
```

Comment:
I want to say "no" ;-)
The optimal size of a label on one line is always the same whatever the available size (even if it is smaller).

Reply:
hm currently no, StringPainter & ImagePainter call Math.min to get minimum value between text/image size and available size. i think we should decide and make it clear

Reply:
Oh, indeed, I didn't saw that. In previous Label, the returned size was always the same and there was no issue. If the widget optimal size is bigger than the constraint, its will probably exceed its final bounds, that's all.

Reply:
yes that's right, i copy/pasted this behavior from ImagePainter because i thought it was nice
ok so this min() shouldn't be done anymore?

Reply:
I think it is useless, yes.

Reply:
fixed